node-jose/index.d.ts

```diff
@@ -0,0 +1,1380 @@
+         * using the Compact serialization (`true`) or the JSON
+         * serialization (`false`).
+         *
+         * @type {boolean}
+         * @memberOf Encrypter
+         * @readonly
+         */
+        compact: boolean;
+
+        /**
+         * Indicates the format the output of this encryption generator takes.
+         *
+         * @type {string}
+         * @memberOf Encrypter
+         * @readonly
+         */
+        format: string;
+
+        /**
+         * The header parameter names that are protected. Protected header fields
+         * are first serialized to UTF-8 then encoded as util.base64url, then used as
+         * the additional authenticated data in the encryption operation.
+         *
+         * @type {string[]}
+         * @memberOf Encrypter
+         * @readonly
+         */
+        protected: string[];
+
+        /**
+         * The global header parameters, both protected and unprotected. Call
+         * {@link Encrypter#protected} to determine which parameters will
+         * be protected.
+         *
+         * @type {object}
+         * @memberOf Encrypter
+         * @readonly
+         */
+        header: object;
+
+        /**
+         * Updates the plaintext data for the encryption generator. The plaintext
+         * is appended to the end of any other plaintext already applied.
+         *
+         * If {data} is a Buffer, {encoding} is ignored. Otherwise, {data} is
+         * converted to a Buffer internally to {encoding}.
+         *
+         * @param {(Buffer|string)} [data] The plaintext to apply.
+         * @param {string} [encoding] The encoding of the plaintext.
+         * @returns {Encrypter} This encryption generator.
+         * @throws {Error} If ciphertext has already been generated.
+         * @memberOf Encrypter
+         */
+        update(data?: Buffer | string, encoding?: string): Encrypter;
+
+        /**
+         * Finishes the encryption operation.
+         *
+         * The returned Promise, when fulfilled, is the JSON Web Encryption (JWE)
+         * object, either in the Compact (if {@link Encrypter#compact} is
+         * `true`) or the JSON serialization.
+         *
+         * @param {(Buffer|string)} [data] The final plaintext data to apply.
+         * @param {string} [encoding] The encoding of the final plaintext data (if any).
+         * @returns {(Promise<string|object>)} A promise for the encryption operation.
+         * @throws {Error} If ciphertext has already been generated.
+         * @memberOf Encrypter
+         */
+        final(data?: Buffer | string, encoding?: string): Promise<string | object>;
+    }
+}
+
+/**
+ * Tools for creating and manipulating keys and keystores.
+ */
+export module JWK {
+
+    /**
+     * The following constants are used to select the different modes for
+     * operation, their names are self-explanatory.
+     */
+    const MODE_DECRYPT: string;
```

Comment:
these should probably be `const MODE_DECRYPT: "MODE_DECRYPT";` and then have a union type tying them together for later use: `type Mode = MODE_DECRYPT | MODE_ENCRYPT | ...`